### ISSUE TITLE
[gandi provider] Add support for new Gandi API (dns.api.gandi.net)

### ISSUE THREAD

**Issue by Ezwen** [AnalogJ/lexicon#171]
The new Gandi website comes with a new API available at https://dns.api.gandi.net

New Gandi accounts can only use this API, and not the previous one. Therefore it would be great  to add support for this new Gandi API to lexicon for such users :)

thanks for your work on Lexicon!

**Comment by AnalogJ** [comment]
Hey @Gwend4l 

Thanks for your interest. It looks like @sumpfralle and @adferrand are already working on that in #162 

Once their work is complete, you should be able to select the Live/REST api using a command line flag.

**Comment by Gwend4l** [comment]
Indeed, I had not noticed, thanks for pointing that out! Looking forward to the merge :)

**Comment by adferrand** [comment]
I think that issue can be closed, as we implemented the gandi provider on top of the rest api in #247.

**Comment by mt7479** [comment]
@adferrand Is it possible to specify the protocol with an environment variable ? I tried the following with no luck.

```
export PROVIDER="gandi"
export LEXICON_GANDI_TOKEN="123"
export LEXICON_GANDI_PROTOCOL="rest"

export LEXICON_GANDI_AUTH_TOKEN="123"
export LEXICON_GANDI_API_PROTOCOL="rest"
```

I patched my dehydrated hook and included the necessary command line parameters but environment variables would the nice though.

**Comment by adferrand** [comment]
In fact only options like --auth-* can be specified with environment variables. But protocol is specified with --api-protocol. I named it like this because this option is related to the provider itself, not to the authentication.

@AnalogJ, I think it would be useful to specify any option by environment variables.

**Comment by mt7479** [comment]
Ah, now i understand. It would indeed be nice have this at least for this specific provider, not sure if any other would benefit as well.

**Comment by nh2** [comment]
> I patched my dehydrated hook and included the necessary command line parameters

@mt7479 Can you share your patches? I'm facing the problem

```
Exception: Failed to authenticate: '<Fault 510150: 'Error on object : OBJECT_ACCOUNT (CAUSE_NORIGHT) [Invalid API key]'>'
```

and suspect that it'll go away by using the REST API.

**Comment by mt7479** [comment]
@nh2 Nice hack! 
I wasn't as smart and simply hard-coded all the needed options in the hook file. Including removing the $PROVIDER Variable :-D

**Comment by adferrand** [comment]
PR is on the go to avoid the hacks ^^

**Comment by adferrand** [comment]
Fixed by #302. Now protocol can be set using LEXICON_GANDI_API_PROTOCOL env variable.
